### PR TITLE
List in-bounds pins first

### LINKED ISSUE #181: Sort pins by visiblity in sidebar
We could enhance the pin list in the sidebar by prioritizing pins that are currently visible to the user based on current image bounds and zoom level. Related to #176

### PR DESCRIPTION
**Pull request by naglepuff**
Resolves #181 

This PR adds the map bounds to the store and a new computed property to the pin list component. These additions are used to sort the pins that are within the map bounds to the top of the list in the sidebar. As a user zooms/pans around the maps, the pins that become visible jump up the list.

To do: 

- [x] Factor zoom level into sorting function

### CODE REVIEW

File: web/src/components/PinList.vue
Comment (waxlamp) [inline web/src/components/PinList.vue]: Looking at the source for `postGisToPoint()`, it doesn't look like it truly returns `Point | undefined`; all the error conditions lead to throwing an exception.

I bring it up because this code doesn't look correct to me--if `postGisToPoint()` were to fail, you wouldn't want to continue from this point using (0, 0). I think the practical effect is to crash the program, but this code falsely implies that it has a chance of continuing from this point with bad data. I think it would be best to fix up the types for that function, and remove the `||` part from this line.

File: web/src/components/PinList.vue
Comment (waxlamp) [inline web/src/components/PinList.vue:73]: Is there a GeoJS function that does this check for you? It seems like the kind of thing that would exist as a utility function, but maybe not.

File: web/src/components/PinList.vue
Comment (waxlamp) [inline web/src/components/PinList.vue]: For readability, you may want to invert this test:

```suggestion
        && zoom.value <= pinMaxZoom + 2
```

**Review by waxlamp — APPROVED** [review]
Looks good! If you accept my suggestions, no need for another review. Otherwise, you can make changes and request another review.

File: web/src/components/PinList.vue
Comment (naglepuff) [reply web/src/components/PinList.vue]: 4fc958a

File: web/src/components/PinList.vue
Comment (naglepuff) [reply web/src/components/PinList.vue:73]: I've asked the geojs experts, but while waiting for a response the best thing I could find on my own was the `boxSearch` [function](https://opengeoscience.github.io/geojs/apidocs/geo.feature.html#boxSearch). We could call this function, which is a member of `pinFeature` to get a list of all pins within a given box. We could then compare the pin in question with the list to see if its within the box. There doesn't seem to be a straightforward function on `geo.map` that helps answer the question: "is this point in bounds?"

I think the current method of sorting the list of pins based on current bounds is more straightforward than getting a list of currently visible pins and comparing it to the list of all pins.

**Pull request merged**
